infrastructure/src/main/java/org/wso2/carbon/testgrid/infrastructure/providers/AWSProvider.java

```diff
@@ -0,0 +1,94 @@
+
+    @Override
+    public String getProviderName() {
+        return "AWS Provider";
+    }
+
+    @Override
+    public boolean canHandle(Infrastructure infrastructure) {
+        //Check if scripts has a cloud formation script.
+        boolean isCloudFormation = false;
+        for (Script script : infrastructure.getScripts()) {
+            if (script.getScriptType().equals(Script.ScriptType.CLOUD_FORMATION)) {
+                isCloudFormation = true;
+            }
+        }
+        //return true only if it has aws node specs or has cloud formation script.
+        return ((infrastructure.getProviderType().equals(Infrastructure.ProviderType.AWS)) && isCloudFormation);
+    }
+
+    @Override
+    public Deployment createInfrastructure(Infrastructure infrastructure, String infraRepoDir)
+            throws TestGridInfrastructureException {
+        AWSDeployer awsDeployer = new AWSDeployer(TestGridConstants.AWS_ACCESS_KEY, TestGridConstants.AWS_SECRET_KEY);
+        awsDeployer.init(infrastructure);
+        try {
+            for (Script script : infrastructure.getScripts()) {
+                if (script.getScriptType().equals(Script.ScriptType.CLOUD_FORMATION)) {
+                    //assumption : Only one CF script will be there
+                    return awsDeployer.createInfrastructure(script, infraRepoDir);
+                }
+            }
+            String errorMessage = "No CloudFormation Script found in script list";
+            throw new TestGridInfrastructureException(errorMessage);
+        } catch (InterruptedException e) {
+            String errorMessage = "Error occured while waiting for CloudFormation Stack creation";
+            throw new TestGridInfrastructureException(errorMessage, e);
+        } catch (IOException e) {
+            String errorMessage = "Error occured while Reading CloudFormation script";
+            throw new TestGridInfrastructureException(errorMessage, e);
+        }
+    }
+
+    @Override
+    public boolean removeInfrastructure(Infrastructure infrastructure, String infraRepoDir)
+            throws TestGridInfrastructureException {
+        AWSDeployer awsDeployer = new AWSDeployer(TestGridConstants.AWS_ACCESS_KEY, TestGridConstants.AWS_SECRET_KEY);
+        awsDeployer.init(infrastructure);
+        try {
+            for (Script script : infrastructure.getScripts()) {
+                if (script.getScriptType().equals(Script.ScriptType.CLOUD_FORMATION)) {
+                    //assumption : Only one CF script will be there
+                    return awsDeployer.destroyInfrastructure(script);
+                }
+            }
+            String errorMessage = "No CloudFormation Script found in script list";
+            throw new TestGridInfrastructureException(errorMessage);
+        } catch (InterruptedException e) {
+            String errorMessage = "Error while waiting for CloudFormation stack to destroy";
+            throw new TestGridInfrastructureException(errorMessage);
```

Comment:
It would be better to include the exception object (e) in the wrapping exception.